www/lib/lib.php

```diff
@@ -767,20 +767,37 @@ public function getInfo($id){
 			$this->ptzControl = new cameraPtz($this);
 		}
 	}
-	public function checkConnection(){ 
+	public function checkConnection() {
+		 //FIXME: Currently only supports RTSP. MJPEG is currently untested
 		ini_set('default_socket_timeout', 1);
-		#needs server to check for RTSP //// $paths['rtsp']  = 'http://'.((empty($this->info['rtsp_username'])) ? '' : $this->info['rtsp_username'].':'.$this->info['rtsp_password'].'@').$this->info['ipAddr'].':'.$this->info['port'].$this->info['rtsp'];
-		$paths['mjpeg'] = 'http://'.((empty($this->info['rtsp_username'])) ? '' : $this->info['rtsp_username'].':'.$this->info['rtsp_password'].'@').((empty($this->info['ipAddrMjpeg'])) ? $this->info['ipAddr'] : $this->info['ipAddrMjpeg']).':'.$this->info['portMjpeg'].$this->info['mjpeg_path'];
-		$paths['http'] = 'http://'.((empty($this->info['rtsp_username'])) ? '' : $this->info['rtsp_username'].':'.$this->info['rtsp_password'].'@').((empty($this->info['ipAddrMjpeg'])) ? $this->info['ipAddr'] : $this->info['ipAddrMjpeg']);
-		foreach($paths as $type => $path){
-			$headers = @get_headers($path);
-			$contents = @file_get_contents($path);
-			if (!$headers) { $this->info['connection_status'][$type] = 'F'; continue; }
-			preg_match("/([0-9]{3})/", $headers[0], $response_code);
-			$this->info['connection_status'][$type] = ($response_code[0]=='200') ? 'OK' : $response_code[0];
+
+		$path = "";
+		$args = "";
+
+		switch($this->info['protocol'])  {
+			case 'IP-RTSP':
+				$path = 'rtsp://'.((empty($this->info['rtsp_username'])) ? '' : $this->info['rtsp_username'].':'.$this->info['rtsp_password'].'@').$this->info['ipAddr'].':'.$this->info['port'].$this->info['rtsp'];
+				$args = array("-rtsp_flags +prefer_tcp", "-rtsp_transport tcp", "-rtsp_transport tcp")[$this->info['rtsp_rtp_prefer_tcp']];
```

Comment:
Fixing the UDP case.

```suggestion
				$rtp_args_menu = array("-rtsp_flags +prefer_tcp", "-rtsp_transport tcp", "-rtsp_transport +udp+udp_multicast")
				$args = $rtp_args_menu[$this->info['rtsp_rtp_prefer_tcp']];
```
Suggesting the split into two lines because the last bit which selects the entry was hard to notice.